contrib/example-backend/exporttemplate/index.go

```diff
@@ -0,0 +1,93 @@
+/*
+Copyright 2023 The Kube Bind Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package exporttemplate
+
+// TODO by namespace
+// TODO cached client
+// TODO find a better name or reorganize into different packages
+import (
+	"context"
+	"fmt"
+
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	crd "k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/rest"
+
+	"github.com/kube-bind/kube-bind/contrib/example-backend/apis/examplebackend/v1alpha1"
+	templates "github.com/kube-bind/kube-bind/contrib/example-backend/client/clientset/versioned"
+)
+
+type Index struct {
```

Comment:
not sure I can follow this construct. What is it?

Reply:
The backend needs to know which CRD get which claimedresources. I stored them in APIServiceExportTemplates. `Index` lets the backend get all exported CRDs (those that are referenced by a APIServiceExportTemplate), and to look up the template for a CRD so it can attach the claimed resources